qa-tests-backend/src/test/groovy/Enforcement.groovy

```diff
@@ -698,6 +698,12 @@ class Enforcement extends BaseSpecification {
         assert lockProcessBaselines.size() ==  1
         assert  lockProcessBaselines.get(0).getElementsList().
                 find { it.element.processName.equalsIgnoreCase("/usr/sbin/nginx") } != null
+
+        // Wait for the baseline to be synced with Sensor.
+        // If this tests flakes again log statements were added in Central and Sensor to investigate if a
+        // baseline was synced with sensor at a specific point in time.
+        sleep 10000
```

Comment:
nitpick: it is usually better to avoid static sleep. It will extend the test execution time when there is no need for it. However, in other cases, wait time will not be sufficient, and we will still get a failed test. If possible, it is better to wait for some event.

I see that we have additional logs now. Can we use `orchestrator.getPodLog` to check if the expected event happened?

I know we have similar logic in non-groovy tests, but I didn't see it in groovy tests. 😞

Reply:
Maybe we could do something like this:

In a retry loop
Refetch the baseline until the `UserLockTimestamp` is set.

That way we wouldn't have to wait on the logs and could remove the sleep entirely for an event driven item.

Reply:
+1
I don't know if we have something similar in groovy tests, but I think I could try to migrate these test cases to Go and use the function there. This would solve it.

I've added a ticket for this, your approach to listen to the logs seems simpler and more straight forward: https://issues.redhat.com/browse/ROX-27486